test/ServerRequestFilter/FilterUsingXForwardedHeadersTest.php

```diff
@@ -368,4 +368,30 @@ public function testOnlyHonorsXForwardedProtoIfValueResolvesToHTTPS(
         $uri             = $filteredRequest->getUri();
         $this->assertSame($expectedScheme, $uri->getScheme());
     }
+
+    public function testPortExtractedFromXForwardedHostHeader(): void
+    {
+        $request = new ServerRequest(
+            ['REMOTE_ADDR' => '192.168.1.1'],
+            [],
+            'http://localhost:80/foo/bar',
+            'GET',
+            'php://temp',
+            [
+                'Host'              => 'localhost',
+                'X-Forwarded-Host'  => 'example.com:4433',
+                'X-Forwarded-Port'  => '4433',
```

Comment:
I think `X-Forwarded-Port` should be different, to make sure that the header contents take priority over `X-Forwarded-Host`

Reply:
```suggestion
                'X-Forwarded-Port'  => '8433',
```